core/src/main/java/org/apache/druid/java/util/common/guava/ParallelMergeCombiningSequence.java

```diff
@@ -121,8 +123,9 @@ public <OutType> Yielder<OutType> toYielder(OutType initValue, YieldingAccumulat
     if (inputSequences.isEmpty()) {
       return Sequences.<T>empty().toYielder(initValue, accumulator);
     }
-
-    final BlockingQueue<ResultBatch<T>> outputQueue = new ArrayBlockingQueue<>(queueSize);
+    // we make final output queue larger than the merging queues so if downstream readers are slower to read there is
+    // less chance of blocking the merge
+    final BlockingQueue<ResultBatch<T>> outputQueue = new ArrayBlockingQueue<>(4 * queueSize);
```

Comment:
hmm, yeah I suppose they do have a bit of a complicated interaction... that setting applies to input waiting to be merged and this one applies to the amount of output that has been merged and is waiting by downstream stuff to be picked up. That said, i'm not really sure what to do instead, and I wanted to increase specifically the amount of merged output to be available so that slow readers don't cause pushback everywhere else.

Reply:
Yeah, I also didn't find a good way to compare and harmonize the two limits. I think this fine since the limit is still conservative here